internal/resource/http/http.go

```diff
@@ -0,0 +1,172 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+func NewHandler(
+	client pbresource.ResourceServiceClient,
+	registry resource.Registry,
+	parseToken func(req *http.Request, token *string),
+	logger hclog.Logger) http.Handler {
+	mux := http.NewServeMux()
+	for _, t := range registry.Types() {
+		// Individual Resource Endpoints.
+		prefix := strings.ToLower(fmt.Sprintf("/%s/%s/%s/", t.Type.Group, t.Type.GroupVersion, t.Type.Kind))
+		logger.Info("Registered resource endpoint", "endpoint", prefix)
+		mux.Handle(prefix, http.StripPrefix(prefix, &resourceHandler{t, client, parseToken, logger}))
+	}
+
+	return mux
+}
+
+type writeRequest struct {
+	// TODO: Owner.
```

Comment:
Good catch, I think I need more input regarding this. 
@analogue 
Could you add some context here please? Thank you!

Reply:
@wangxinyi7 Take some time to go over all the possible scenarios in the unit test for the gRPC endpoint @ https://github.com/hashicorp/consul/blob/9b540e29bcc2ac87030529f3d5545184f91c30e9/agent/grpc-external/services/resource/write_test.go which is the delegate for the http endpoint.

Given that, let's meet (you, me, poonam) to hammer out all the acceptance criteria for the ticket (success, failures, edge cases, etc) since the ticket only has a single happy path and not much else.

Reply:
Calendar invitation was sent out. Thanks @analogue !

Reply:
Feel free to take a stab at writing the acceptance criteria before we meet if want to 😄 .